src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/RSAPkcs1X509SignatureGenerator.cs

```diff
@@ -39,7 +39,7 @@ internal static PublicKey BuildPublicKey(RSA rsa)
                 //
                 // This is due to one version of the ASN.1 not including OPTIONAL, and that was
                 // the version that got predominately implemented for RSA. Now it's convention.
-                new AsnEncodedData(oid, new byte[] { 0x05, 0x00 }),
+                new AsnEncodedData(oid, stackalloc byte[] { 0x05, 0x00 }),
```

Comment:
This allocations in https://github.com/dotnet/runtime/blob/5078dd61a81c88dbdd131a77b8148b4dee3803fa/src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedData.cs#L132 anyway. So maybe use a `static readonly byte[]` and pass this in?

Reply:
It would still allocate.

Reply:
Yes, but only once and not for every call to `BuildPublicKey` (don't know how often this internal method will be called).

I mean what's the point of `stackalloc` here when later there's a call to `ToArray`?
It's still better than havingn the `new byte[]` here, as when the implementation of `AsnEncodedData` can/will change this can profit for free of this.

ROS<byte> doesn't make it better also.

Reply:
Look at the RawData setter used by the ctor that takes a byte[].

There's no benefit to having a static byte[] over a span,  so why pay for the field.

Reply:
Ah, this sets the value via property, not via the field -- missed that. Thanks!